### ISSUE TITLE
[Bug] MSAL 4.18.0 Returns Default Set of Scopes on First Call

### ISSUE THREAD

**Issue by phil4business** [AzureAD/microsoft-authentication-library-for-dotnet#2041]
**Which Version of MSAL are you using ?**
MSAL 4.18.0

**Platform**
net472

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [x] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - please describe;

**Is this a new or existing app?**
The app is in production, and I have upgraded to a new version of MSAL. Upgrade from 4.7.1 to 4.18.0. I've tested all intermediate versions as well.

**Repro**

```csharp
public async Task<AuthenticationResult> AcquireATokenFromCacheOrUserPasswordFlowAsync(IEnumerable<String> scopes, PSCredential credential)
        {
            AuthenticationResult result = null;
            var accounts = await App.GetAccountsAsync();

            if (accounts.Any(account => account.Username == credential.UserName))
            {
                result = await App.AcquireTokenSilent(scopes, accounts.FirstOrDefault(account => account.Username == credential.UserName))
                    .ExecuteAsync();
            }
            if (result == null)
            {
                result = await await App.AcquireTokenByUsernamePassword(scopes, credential.UserName, credential.Password)
                .ExecuteAsync();
            }

            return result;
        }
```

**Expected behavior**
MSAL acquires a token with the custom scope.
Previous version returned a token with only the custom (e.g.```api://{audienceId}/admin```) scope.

**Actual behavior**
MSAL acquires a token with scopes ```User.Read```, ```profile```, ```openid```, and ```email```. These scopes are always returned no matter what I provide for the scopes parameter. A subsequent call to ```AcquireTokenSilent(IEnumerable<string> scopes, IAccount account)``` then returns another (new) access token for the custom scope only.
These default scopes are always returned no matter what I provide for the scopes parameter.

**Possible Solution**
N/A

**Additional context/ Logs / Screenshots**
N/A

**Comment by jmprieur** [comment]
Which `scopes` are you asking for? (something ending in `.default` ? 

I believe that this has nothing to do with MSAL, but with app registration. I suppose that someone has done some admin consent on resources, which are already approved, and therefore all these approved scoped would be returned in the token.

Is it a problem for you?

**Comment by phil4business** [comment]
```Scopes = new string[] { "User.Read", "api://{audienceId}/Admin" }``` were all that I was requesting. In previous versions a token with only the api://{audienceId}/Admin scope was returned.

> I suppose that someone has done some admin consent on resources, which are already approved, and therefore all these approved scoped would be returned in the token.

Yes, the app registration has 2 api permissions approved. Microsoft Graph/User.Read and the custom api permission (granted with admin consent). The app registration has not changed, but reverting the library to v 4.17.0. reverts the behaviour.

**Comment by jmprieur** [comment]
@phil4business : Something I've noticed also is that the scopes you ask for involve two resources (Graph and your API).
In general, Azure AD gets token for only one resource at the time (but user.read is a bit special as it's also the profile)

you might want to do do 2 calls:
- AcquireTokenXXX("api://{audienceId}/Admin")
- then AcquireTokenSilent("User.Read")

**Comment by phil4business** [comment]
>>What are you expecting from us? getting back to the previous behavior? may I ask why the new behavior is a problem for you?
Yes going back to the previous 1-call behaviour would be preferred but:

I did not realize I was effectively requesting access to two different apis with User.Read and /Admin. I thought User.Read was some kind of "generic" permission. Now that you've pointed it out, this behaviour totally makes sense.

Is there an example application or a place in the docs where this is talked about? From what I've found, there are examples where a token is retrieved with a scope for 1 endpoint, or multiple scopes for 1 endpoint (including incremental consent) but no examples where the application calls multiple different resources. I assumed I could request multiple resources for my application because I am only asking Azure AD for the token. Azure AD would then build and return 1 token with all the scopes required.
